flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/ConsumableNotifyingResultPartitionWriterDecorator.java

```diff
@@ -119,6 +121,11 @@ public void fail(Throwable throwable) {
 		partitionWriter.fail(throwable);
 	}
 
+	@Override
+	public CompletableFuture<?> isAvailable() {
+		return AvailabilityProvider.AVAILABLE;
```

Comment:
Hmm? Doesn't it make all rest of the code effectively an unused/dead code?

Reply:
Thanks for finding this critical issue. It was my careless to take it simple as other tests way. 
Actually the `ConsumableNotifyingResultPartitionWriterDecorator` is used by `RecordWriter` in practice, so we should create this decorator in new added test `RecordWriterTest#testIsAvailableOrNot` to cover this issue. And I already fixed it.